src/main/java/org/breedinginsight/brapps/importer/daos/BrAPIObservationUnitDAO.java

```diff
@@ -114,8 +123,16 @@ public List<BrAPIObservationUnit> getObservationUnitsForTrialDbId(@NotNull UUID
         observationUnitSearchRequest.trialDbIds(List.of(trialDbId));
 
         ObservationUnitsApi api = brAPIEndpointProvider.get(programDAO.getCoreClient(program.getId()), ObservationUnitsApi.class);
-        return brAPIDAOUtil.search(api::searchObservationunitsPost,
+        List<BrAPIObservationUnit> observationUnits = brAPIDAOUtil.search(api::searchObservationunitsPost,
                 api::searchObservationunitsSearchResultsDbIdGet,
                 observationUnitSearchRequest);
+        if( withGID ){
+            for (BrAPIObservationUnit observationUnit: observationUnits) {
+                String germplasmDbId = observationUnit.getGermplasmDbId();
+                BrAPIGermplasm germplasm = this.germplasmService.getGermplasmByDBID(program.getId(), germplasmDbId).get() ;
+                observationUnit.putAdditionalInfoItem("gid", germplasm.getAccessionNumber());
```

Comment:
Can we add gid to BrAPIAdditionalInfoFields constants and use that here?

Reply:
Thanks.
Done.